index.js

```diff
@@ -501,17 +506,29 @@ module.exports = class Protomux {
     const end = state.end
     let remoteId = c.uint.decode(state)
 
+    let waiting = null
+
     while (state.end > state.start) {
       const len = c.uint.decode(state)
       if (len === 0) {
         remoteId = c.uint.decode(state)
         continue
       }
       state.end = state.start + len
-      this._decode(remoteId, state)
+      // if batch contains more than one message, cork it so we reply back with a batch
+      if (end !== state.end && waiting === null) {
+        waiting = []
+        this.cork()
```

Comment:
Sanity check: can `_onbatch` be called again before `Promise.all(waiting).then(this._uncorkBound, this._safeDestroyBound)` has resolved?

If so, I think that breaks the logic, because the `cork` state isn't managed per batch, but for the entire Protomux object. So the `uncork` base case of `--this.corked === 0` might trigger only after a lot of batches

Reply:
ya thats a feature, it'll keep buliding it up until it hits corked === 0 or the max batch size

Reply:
we def might wanna add some additional logic, like a max timeout also etc or to reduce the max batch size, which right now is set to 8mb i believe